File: WoofIOS/SwiftPackages/DTO/Sources/DTO/Dog/DogsPageDTO.swift
import Foundation

public struct DogsPageDTO: Codable, Equatable {
    public let results: [DogDTO]
    public let offset: Int
    public let total: Int

    enum CodingKeys: String, CodingKey {
        case results
        case offset
        case total
    }

    public init(results: [DogDTO] = [], offset: Int, total: Int) {
        self.results = results
        self.offset = offset
        self.total = total
    }
}

File: WoofAndroid/dto/src/main/kotlin/com/perrystreet/woof/dto/dog/DogsPageDTO.kt
package com.perrystreet.woof.dto.dog

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class DogsPageDTO(
    @Json(name = "results") val results: List<DogDTO> = emptyList(),
    @Json(name = "offset") val offset: Int,
    @Json(name = "total") val total: Int,
)
